doc the weblate workflow for Volto (#1957)
plone/documentation: docs/i18n-l10n/contributing-translations.md

Maintainers will periodically review the pull request that Weblate creates automatically, and merge it.

### Wewblate workflow in Volto

> Volto does some checks in all pull-requests that require a changelog entry to be present and the po files to have a given formatting.

Reviewer: ```suggestion
For all pull requests, Volto's CI checks for both the presence of a change log entry and that the `.po` files used in translations comply with a standard format.
```

> Weblate doesn't create such changelog entry and its po style guide does not match the one of Volto, so to merge the changes coming to weblate, a maintainer has to do the following:

Reviewer: When you say "maintainer", do you mean "Volto Team" member? Otherwise, I'm afraid most people will assume that is only @sneridagh, and it would be good to spread the workload.

```suggestion
Weblate doesn't create a change log entry, and its `.po` file style guide does not match Volto's.
This means that a maintainer has to perform the following tasks to merge the changes coming from Weblate.
```

Author: I will do the relevant checks whenever there is a weblate PR, but I wanted a more general word, and that's why I wrote "Maintainer", but yes, "Volto Team member" could be a proper replacement.

> - Clone the `translations-18.x.x` branch from the Volto repository
> - Add a changelog file in the `news` folder with the relevant entry
> - Run `make i18n` to reformat the po files.
> - Push back the changes to GitHub and wait to the CI checks to success.
> - Merge the pull request when the checks are green.

Reviewer: ```suggestion
- Checkout the `translations-18.x.x` branch from the Volto repository.
- Add a change log file in the `packages/volto/news` folder.
- Run `make i18n` to reformat the `.po` files.
- Commit and push the changes to GitHub.
- When all CI checks succeed, then merge the pull request.
```

Maintainers will periodically review the pull request that Weblate creates automatically, and merge it.

### Wewblate workflow in Volto

For all pull requests, Volto's CI checks for both the presence of a change log entry and that the `.po` files used in translations comply with a standard format.

Weblate doesn't create a change log entry, and its `.po` file style guide does not match Volto's.

> This means that a maintainer has to perform the following tasks to merge the changes coming from Weblate.

Reviewer: ```suggestion
This means that a Volto Team member has to perform the following tasks to merge the changes coming from Weblate.
```
